avocado/core/runners/utils/messages.py

```diff
@@ -0,0 +1,198 @@
+    :param returncode: exit status of runner
+    :return: finished message
+    :rtype: dict
+    """
+    message = {'result': result}
+    if fail_reason is not None:
+        message['fail_reason'] = fail_reason
+    if returncode is not None:
+        message['returncode'] = returncode
+    return _prepare_message('finished', message)
+
+
+def get_running_message():
+    """Creates running message without any additional info.
+
+    :return: running message
+    :rtype: dict
+    """
+    return _prepare_message('running')
+
+
+def get_log_message(message):
+    """Creates log message with  all necessary information.
+
+    :param message: log to be sent. The str will be encoded with utf-8.
+    :type message: str, bytes
+    :return: log message
+    :rtype: dict
+    """
+    return _get_running_message(message, 'log')
+
+
+def get_stdout_message(message):
+    """Creates stdout message with  all necessary information.
+
+    :param message: data to be sent. The str will be encoded with utf-8.
+    :type message: str, bytes
+    :return: stdout message
+    :rtype: dict
+    """
+    return _get_running_message(message, 'stdout')
+
+
+def get_stderr_message(message):
+    """Creates stderr message with  all necessary information.
+
+    :param message: data to be sent. The str will be encoded with utf-8.
+    :type message: str, bytes
+    :return: stderr message
+    :rtype: dict
+    """
+    return _get_running_message(message, 'stderr')
+
+
+def get_whiteboard_message(message):
+    """Creates whiteboard message with  all necessary information.
+
+    :param message: data to be sent. The str will be encoded with utf-8.
+    :type message: str, bytes
+    :return: whiteboard message
+    :rtype: dict
+    """
+    return _get_running_message(message, 'whiteboard')
+
+
+class RunnerLogHandler(logging.Handler):
+
+    def __init__(self, queue, message_type):
+        """
+        Runner logger which will put every log to the runner queue
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param message_type: type of the log
+        :type message_type: string
+        """
+        super().__init__()
+        self.queue = queue
+        self.message_type = message_type
+
+    def emit(self, record):
+        msg = self.format(record)
+        self.queue.put(_get_running_message(msg, self.message_type))
+
+
+class StreamToQueue:
+
+    def __init__(self,  queue, message_type):
+        """
+        Runner Stream which will transfer every  to the runner queue
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param message_type: type of the log
+        :type message_type: string
+        """
+        self.queue = queue
+        self.message_type = message_type
+
+    def write(self, buf):
+        self.queue.put(_get_running_message(buf, self.message_type))
+
+    def flush(self):
+        pass
```

Comment:
And those two methods could be part of the `GenericMessage` class that I mentioned. Not sure if makes sense, just brainstorming.